lib/components/FileAttachment/index.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import "@uppy/core/dist/style.css";
+import "@uppy/progress-bar/dist/style.css";
+import { saveAs } from "file-saver";
+import { isEmpty } from "ramda";
+
+import directUploadsApi from "apis/direct_uploads";
+import { DIRECT_UPLOAD_ENDPOINT } from "common/constants";
+import Button from "components/Common/Button";
+import useUppyUploader from "hooks/useUppyUploader";
+
+import Attachments from "./Attachments";
+import {
+  DEFAULT_UPPY_CONFIG,
+  DROP_DOWN_OPTIONS_KEYS,
+  UPPY_UPLOAD_CONFIG,
+} from "./constant";
+
+import Progress from "../Editor/MediaUploader/Progress";
+
+const FileAttachment = ({
+  endpoint = DIRECT_UPLOAD_ENDPOINT,
+  onRename = _ => {},
+  onDelete = _ => {},
+  onError = _ => {},
```

Comment:
There callbacks are not very useful IMO. We can remove them. The `attachments` and `onAttach` props can handle all the scenarios. All errors can be handled internally.

Reply:
@AbhayVAshokan for the `onAttach` prop, what would be the signature of this function? and logic for manipulating  attachments e.g. `removing record from the state when an attachment` should it be handled by the host application or it should be handled by the `FileAttachment` component ?

Reply:
@saroj990 good question. This function needs to be called on creating or destroying an attachment. In that case, let's call it `onChange`.

It will be of this format: `(attachments) => {}`. It accepts a list of attachment meta data. Each element will be an object with the following keys: content_type, filename, signed_id, and url.